WalletWasabi/Tor/Control/TorControlClientFactory.cs

```diff
@@ -36,7 +37,7 @@ public TorControlClientFactory(IRandom? random = null)
 	/// <summary>Connects to Tor Control endpoint and authenticates using safe-cookie mechanism.</summary>
 	/// <seealso href="https://gitweb.torproject.org/torspec.git/tree/control-spec.txt">See section 3.5</seealso>
 	/// <exception cref="TorControlException">If TCP connection cannot be established OR if authentication fails for some reason.</exception>
-	public async Task<TorControlClient> ConnectAndAuthenticateAsync(IPEndPoint endPoint, string cookieString, CancellationToken cancellationToken = default)
+	public async Task<TorControlClient> ConnectAndAuthenticateAsync(EndPoint endPoint, string cookieString, CancellationToken cancellationToken = default)
```

Comment:
Because docker uses non-IP end points to connect to Tor and they need that.

Reply:
Quote from the description:

> `TorControlClientFactory` now takes an `EndPoint` instead of only an IP, as wabisabi client / Tor are often running in docker containers which use DNS names (e.g. `tor:9050`)

Reply:
It turns out that the TcpCllient usage inside of wasabi does not support a dns endpoint and so I had to resolve the ip from host and provide it with an ip endpoint upfront. 

Will revert this change.